src/audiomixerboard.cpp

```diff
@@ -1305,6 +1307,49 @@ void CAudioMixerBoard::SetAllFaderLevelsToNewClientLevel()
     }
 }
 
+void CAudioMixerBoard::AutoAdjustAllFaderLevels()
+{
+    QMutexLocker locker ( &Mutex );
+
+    for ( int i = 0; i < MAX_NUM_CHANNELS; ++i )
+    {
+        // only apply to visible faders (and not to my own channel fader)
+        if ( vecpChanFader[i]->IsVisible() && ( true /* i != iMyChannelID */ ) )
+        {
+            // get averaged level meter value (range 0...8)
+            float levelMeter = vecAvgLevels[i];
+
+            // map meter output level to deciber (range -50dB...0dB)
+            // (invert CStereoSignalLevelMeter::CalcLogResultForMeter)
+            float leveldB = levelMeter *
+                ( UPPER_BOUND_SIG_METER - LOW_BOUND_SIG_METER ) /
+                NUM_STEPS_LED_BAR + LOW_BOUND_SIG_METER;
+
+            // map decibels to linear level
+            // level = pow ( 10.0f, level / 20.0f );
+
+            // do not adjust channels with zero level to full level since
+            // the channel might simply be silent at the moment
+            if ( leveldB < -40.0f )
+            {
+                // specify target level
+                float targetdBLevel = -30.0f;
```

Comment:
This is a const by the look of it, right?  Definitely shouldn't be assigned this deep inside the method (okay, compiler probably sorts that out but style-wise, it's best defined somewhere as a constant).

Reply:
I will introduce constants for both the threshold level and the target level on `global.h`.